src/main/scala/com/gu/fastly/PurgedEventWriter.scala

```diff
@@ -0,0 +1,26 @@
+package com.gu.fastly
+
+import com.amazonaws.services.kinesis.model.Record
+import com.amazonaws.services.kinesis.producer.{ KinesisProducer, UserRecordResult }
+import com.google.common.util.concurrent.{ FutureCallback, Futures, ListenableFuture }
+
+import scala.concurrent.ExecutionContext.Implicits.global
+import scala.concurrent.{ Future, Promise }
+
+object PurgedEventWriter {
+
+  def addRecords(records: List[Record], producer: KinesisProducer): Future[List[UserRecordResult]] = Future.traverse(records)(r =>
+    producer.addUserRecord("fastly-cache-purger-events", r.getPartitionKey, r.getData).asScala)
+
+  implicit class RichListenableFuture[T](lf: ListenableFuture[T]) {
```

Comment:
This is from here:
https://stackoverflow.com/questions/18026601/listenablefuture-to-scala-future

It looks like Ophan does something similar:
https://github.com/guardian/ophan/blob/master/shared-lib/src/main/scala/ophan/GuavaFuturable.scala#L34

It's used to convert the Guava ListenableFuture to a Scala Future